### PR TITLE
removing rmbProxy endpoints, RMB, Ygg, and Redis

### PR DESCRIPTION
**Pull request by sameh-farouk**
### Description

as a part of switching to the new RMB implementation, we are refactoring the grid-proxy server to remove the rmb-proxy functionality and any related services such RMB, Ygg and Redis.

### Changes

- [x] Remove rmbproxy endpoints
- [x] Remove rmb, ygg, and redis
- [x] Remove all obsolete configuration
- [x] Update chart, remove RMB Kubernetes service 
- [x] Update swagger docs
- [x] Update Readme

**Removed Services:**
- RMB
- Ygg
- Redis

**Removed options:**
- redis
- substrate
- rmbTimeoutSeconds

**Removed Env Vars:**
- MNEMONICS
- SUBSTRATE
- REDIS
- RMB_TIMEOUT

**Removed Endpoints:**
/twin/*

### Related Issues

https://github.com/threefoldtech/tfgridclient_proxy/issues/291

### Checklist

- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings

### CODE REVIEW

**Review by xmonader — CHANGES_REQUESTED** [review]
please restore everything related to tls/certs

**Comment by sameh-farouk** [comment]
> please restore everything related to tls/certs

@xmonader thanks for your review, requested changes were done in 0474a2a and 3efaf20

however I have these notes, first, I removed this code because
1. It was implemented mainly in the rmbproxy pkg, I thought that maybe was planned for something RMB related, I didn't see it used anywhere but maybe I'm wrong.
2. I understand The helm chart uses Kubernetes certificate management controller, not depend on that code.  
3. I have an impression that the code that serves TLS is not complete, also given that everywhere we start the grid proxy we pass `--no-cert` wasn't sure that it is still relevant.

now I have a question, why do we need this anyway? regarding any future docker-compose projects wouldn't be better to configure an Nginx container to do TLS Offloading instead or use our gateway?

File: Dockerfile [hunk 1]
```diff
@@ -1,50 +1,30 @@
...
-    chmod +x msgbusd
-
 FROM alpine:3.14
 
-RUN apk --update add redis 
-
-COPY --from=builder /src/yggdrasil /usr/bin/yggdrasil
-COPY --from=builder /src/yggdrasilctl /usr/bin/yggdrasilctl
-COPY --from=builder /src/genkeys /usr/bin/genkeys
 COPY --from=builder /src/tfgridclient_proxy/cmds/proxy_server/server /usr/bin/server
```
Comment (xmonader) [inline Dockerfile]: let's user a better name than server

File: README.md [hunk 2]
```diff
@@ -28,11 +28,8 @@ To list all the available tasks for running:
 
 ## Prerequisites
 
-1. A [msgbusd](https://github.com/threefoldtech/rmb_go) instance must be running on the node. This client uses RMB (message bus) to send messages to nodes, and get the responses.
-2. A valid MNEMONICS.
-3. [yggdrasil](https://yggdrasil-network.github.io/installation.html) service running with a valid ip assigned to the MNEMONICS on [polkadot](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Ftfchain.dev.grid.tf%2Fws#/accounts).
-4. Golang compiler > 1.13 to run the grid proxy server.
-5. Postgres database
+1. Golang compiler > 1.13 to run the grid proxy server.
```
Comment (xmonader) [inline README.md]: The compiler shouldn't be a requirement, always link to the release binaries on github, you probably need a separate developer doc e.g wiki/devsetup.md

File: README.md [hunk 3]
```diff
@@ -28,11 +28,8 @@ To list all the available tasks for running:
...
+2. Postgres database
```
Comment (xmonader) [inline README.md]: Only the dsn/connection string correct?

File: README.md [hunk 4]
```diff
@@ -78,17 +71,13 @@ To list all the available tasks for running:
...
 | -postgres-user | postgres username  |
-| -redis | redis url (default `"tcp://127.0.0.1:6379"`)  |
-| -substrate-user | substrate url (default`"wss://tfchain.dev.grid.tf/ws"`)  |
-| -rmb-timeout | timeout for rmb requests (default `30` seconds) |
 | -v | shows the package version |
 
 
 - Then visit `http://localhost:8080/<endpoint>`
 
 ## Production Run
```
Comment (xmonader) [inline README.md:52]: Should be documented before development run

File: README.md [hunk 5]
```diff
@@ -108,7 +97,7 @@ After=network.target
 After=msgbus.service
```
Comment (xmonader) [inline README.md]: Do we still need the msgbusd service? its purpose was proxying messages to zos nodes, which isn't the case anymore, all of our clients can communicate with nodes through relays

File: README.md [hunk 6]
```diff
@@ -142,7 +131,6 @@ EOF
   - domain: the host domain which will generate ssl certificate to.
   - email: the mail used to run generate the ssl certificate.
   - ca: certificate authority server url
```
Comment (xmonader) [inline README.md]: Add example links e.g let's encrypt production and staging for testing.

File: README.md [hunk 7]
```diff
@@ -161,19 +149,13 @@ systemctl daemon-reload
...
 ## Dockerfile
 
-- get public and private key for a yggdrasil configuration
-
 To build & run dockerfile
 
 ```bash
 docker build -t threefoldtech/gridproxy .
-docker run --name gridproxy -e MNEMONICS="" -e SUBSTRATE="wss://tfchain.dev.grid.tf/ws" -e PUBLIC_KEY="5011157c2451b238c99247b9f0793f66e5b77998272c00676d23767fe3d576d8" -e PRIVATE_KEY="ff5b3012dbec23e86e2fde7dcd3c951781e87fe505be225488b50a6bb27662f75011157c2451b238c99247b9f0793f66e5b77998272c00676d23767fe3d576d8" -e POSTGRES_HOST="127.0.0.1" -e POSTGRES_PORT="5432" -e POSTGRES_DB="db" -e POSTGRES_USER="postgres" -e POSTGRES_PASSWORD="password" -e RMB_TIMEOUT="30" --cap-add=NET_ADMIN threefoldtech/gridproxy
+docker run --name gridproxy -e POSTGRES_HOST="127.0.0.1" -e POSTGRES_PORT="5432" -e POSTGRES_DB="db" -e POSTGRES_USER="postgres" -e POSTGRES_PASSWORD="password" --cap-add=NET_ADMIN threefoldtech/gridproxy
```
Comment (xmonader) [inline README.md]: Do we need cap NET_ADMIN? we don't do any trickery with the network interfaces.

File: docs/swagger.json [hunk 8]
```diff
@@ -666,29 +666,6 @@
                 }
             }
         },
-        "/ping": {
```
Comment (xmonader) [inline docs/swagger.json]: why did we remove the healthcheck endpoint?

File: docs/swagger.yaml [hunk 9]
```diff
@@ -745,21 +692,6 @@ paths:
       summary: Show the details for specific node
       tags:
       - GridProxy
-  /ping:
```
Comment (xmonader) [inline docs/swagger.yaml]: Why did we remove the healthcheck endpoint?

File: charts/gridproxy/README.md [hunk 10]
```diff
@@ -30,17 +24,10 @@
...
-  **Note**: EXPLORER_URL, SERVER_IP and REDIS_URL has default values you may not pass them if you want to use the defaults
+  **Note**: EXPLORER_URL, SERVER_IP has default values you may not pass them if you want to use the defaults
 
   ```bash
-  helm install -f values.yaml gridproxy . --set ingress.host="gridproxy.3botmain.grid.tf" --set env.MNEMONICS="" --set env.SUBSTRATE="wss://tfchain.dev.grid.tf/ws" --set env.PUBLIC_KEY="5011157c2451b238c99247b9f0793f66e5b77998272c00676d23767fe3d576d8" --set env.PRIVATE_KEY="ff5b3012dbec23e86e2fde7dcd3c951781e87fe505be225488b50a6bb27662f75011157c2451b238c99247b9f0793f66e5b77998272c00676d23767fe3d576d8" --set env.PEERS="  tls:\\\/\\\/62.210.85.80:39575\\\n   tls:\\\/\\\/54.37.137.221:11129\\\n" --set env.POSTGRES_HOST="127.0.0.1" --set env.POSTGRES_PORT="5432" --set env.POSTGRES_DB="db" --set env.POSTGRES_USER="postgres" --set env.POSTGRES_PASSWORD="password" --set env.RMB_TIMEOUT="30"
+  helm install -f values.yaml gridproxy . --set ingress.host="gridproxy.3botmain.grid.tf" --set env.POSTGRES_HOST="127.0.0.1" --set env.POSTGRES_PORT="5432" --set env.POSTGRES_DB="db" --set env.POSTGRES_USER="postgres" --set env.POSTGRES_PASSWORD="password"
```
Comment (xmonader) [inline charts/gridproxy/README.md]: let's not use 3botmain let's make it gridproxy.dev.grid.tf

File: charts/gridproxy/README.md [hunk 11]
```diff
@@ -30,17 +24,10 @@
 
 - Apply certificate `kubectl create -f prod_issuer.yaml`
 
-- If you want to add more peers add using `--set env.PEERS="  tls:\\\/\\\/62.210.85.80:39575\\\n   tls:\\\/\\\/54.37.137.221:11129\\\n"` add `\\\` as escape characters before each `/`
-
 - Install the chart
 
-  **Note**: EXPLORER_URL, SERVER_IP and REDIS_URL has default values you may not pass them if you want to use the defaults
+  **Note**: EXPLORER_URL, SERVER_IP has default values you may not pass them if you want to use the defaults
```
Comment (xmonader) [inline charts/gridproxy/README.md]: What is EXPLORER_URL?

File: charts/gridproxy/values.yaml [hunk 12]
```diff
@@ -11,22 +11,10 @@ image:
   # tag: "v1.7.0-rc1"
 
   env:          
-    - name: "MNEMONICS"
-      value: ""
     - name: "SERVER_PORT"
       value: ""
     - name: "EXPLORER"
```
Comment (xmonader) [inline charts/gridproxy/values.yaml]: What is that used for? also the docs says EXPLORER_URL not EXPLORER

File: cmds/proxy_server/main.go [hunk 13]
```diff
@@ -26,42 +25,36 @@ const (
...
 	flag.StringVar(&f.postgresUser, "postgres-user", "", "postgres username")
 	flag.StringVar(&f.postgresPassword, "postgres-password", "", "postgres password")
-	flag.StringVar(&f.redis, "redis", "tcp://127.0.0.1:6379", "redis url")
 	flag.BoolVar(&f.version, "v", false, "shows the package version")
-	flag.StringVar(&f.certCacheDir, "cert-cache-dir", CertDefaultCacheDir, "path to store generated certs in")
 	flag.BoolVar(&f.nocert, "no-cert", false, "start the server without certificate")
-	flag.IntVar(&f.rmbTimeoutSeconds, "rmb-timeout", 30, "rmb requests timeout (default 30 sec)")
+	flag.StringVar(&f.domain, "domain", "", "domain on which the server will be served")
+	flag.StringVar(&f.TLSEmail, "email", "", "tmail address to generate certificate with")
+	flag.StringVar(&f.CA, "ca", "https://acme-staging-v02.api.letsencrypt.org/directory", "certificate authority used to generate certificate")
```
Comment (xmonader) [inline cmds/proxy_server/main.go]: move the default to the production one please

**Review by xmonader — CHANGES_REQUESTED** [review]
Add few comments please check

**Comment by xmonader** [comment]
> > please restore everything related to tls/certs
> 
> @xmonader thanks for your review, requested changes were done in [0474a2a](https://github.com/threefoldtech/tfgridclient_proxy/commit/0474a2aa35b1290ffbdfd58b75f70a67a43118fc) and [3efaf20](https://github.com/threefoldtech/tfgridclient_proxy/commit/3efaf20b57477e400b59b9727368108d4dd49b42)
> 
> however I have these notes, first, I removed this code because
> 
> 1. It was implemented mainly in the rmbproxy pkg, I thought that maybe was planned for something RMB related, I didn't see it used anywhere but maybe I'm wrong.

It was implemented for the gridproxy itself, not related to the rmb proxying features. In general given that code for acme client is already solid in go (thanks to many server applications e.g caddy) we can reuse it in our binaries to depend on less and less, so no need to drag in more components e.g caddy or nginx


> 2. I understand The helm chart uses Kubernetes certificate management controller, not depend on that code.

It's true that gridproxy runs on kubernetes -for the time being- which wasn't the case in the beginning where we managed the proxy machines in regular VMs setups, and now we are moving away from that again to docker-compose based setups

> 3. I have an impression that the code that serves TLS is not complete, also given that everywhere we start the grid proxy we pass `--no-cert` wasn't sure that it is still relevant.
> 

It's complete and working, we are only support http-01 flow, also the usage of --no-cert is already to serve use cases where the operator deploys it when kubernetes infrastructure or with a sidecar for TLS.

> now I have a question, why do we need this anyway? regarding any future docker-compose projects wouldn't be better to configure an Nginx container to do TLS Offloading instead or use our gateway?

You can have different kinds of setups e.g on a VM, with container(s), within kubernetes. IMHO, that option with so little code, enabled variety of options for deployment, That's why I wouldn't deleted it

File: Dockerfile (hunk 1 above)
Comment (sameh-farouk) [reply Dockerfile]: @xmonader any suggestions? we didn't yet agree on a new name for the stripped-down version of gridproxy so I'm not sure here.

File: docs/swagger.json (hunk 8 above)
Comment (sameh-farouk) [reply docs/swagger.json]: it was a part of the rmbproxy server. restored now.

**Comment by sameh-farouk** [comment]
> Add few comments please check

@xmonader all non-doc related issues were resolved (except for the server binary name waiting for your reply).
Do you mind handling all the docs-related issues in a separate PR where docs can be focused on and revised? it would be easier also for reviewers. if so I can open an issue for the docs, link it on a new PR and work on it.

File: charts/gridproxy/values.yaml (hunk 12 above)
Comment (sameh-farouk) [reply charts/gridproxy/values.yaml]: this is not used /not exist. I removed that key from the values.yaml

File: charts/gridproxy/README.md (hunk 11 above)
Comment (sameh-farouk) [reply charts/gridproxy/README.md]: not used anywhere on the server or the image. should be removed, also SERVER_IP is not used, should be  SERVER_PORT. I think the docs need a full revision.

File: Dockerfile (hunk 1 above)
Comment (xmonader) [reply Dockerfile]: can be gridproxy or gridrest for now

**Comment by xmonader** [comment]
Let's address the minor docs comments in this PR and do your full review in another issue / PR please

**Comment by sameh-farouk** [comment]
> Let's address the minor docs comments in this PR and do your full review in another issue / PR please

@xmonader done in f4854e5d4f007, please check.

**Comment by xmonader** [comment]
Let's update the branch before merging

**Comment by sameh-farouk** [comment]
> Let's update the branch before merging

@xmonader branch updated

**Review by xmonader — APPROVED** [review]

**Pull request merged**